src/hdmf/common/table.py

```diff
@@ -1046,12 +1044,36 @@ def get(self, arg, index=False, **kwargs):
             arg1 = arg[0]
             arg2 = arg[1]
             return self.table[self.data[arg1], arg2]
-        elif isinstance(arg, slice) or np.issubdtype(type(arg), np.integer):
-            if np.issubdtype(type(arg), np.integer) and arg >= len(self.data):
+        elif np.issubdtype(type(arg), np.integer):
+            if arg >= len(self.data):
                 raise IndexError('index {} out of bounds for data of length {}'.format(arg, len(self.data)))
             ret = self.data[arg]
             if not index:
-                ret = self.table.get(ret, **kwargs)
+                ret = self.table.get(ret, df=df, **kwargs)
+            return ret
+        elif isinstance(arg, slice) or isinstance(arg, (list, np.ndarray)):
+            idx = arg
+
+            # ensure a list/ndarray if indices
+            if isinstance(idx, slice):
+                idx = list(range(*idx.indices(len(self.data))))
```

Comment:
Why must idx be a list? Shouldn't indexing with a slice work just as well?

Reply:
You can't directly index a list with a list, so if we are given a list, it has to be done manually (see line 1063). Putting this here simplified the conditional below. It might be possible to put this check under the if at line 1062.

Reply:
Sorry, I think I wasn't clear. If a slice is given, the code here converts the slice into a list of indices. Most (if not all) data types that a VectorData allows can take a slice as input, so it seems that converting the slice to a list is unnecessary. Also, some data types (like h5py.Dataset) may have specialized functions for handling slices that may be better than converting them into a list of indices. So I question whether these two lines are necessary. I think line 1065 should be able to handle slices.